cartography/intel/aws/sqs.py

```diff
@@ -0,0 +1,140 @@
+            queue.last_modified_timestamp = sqs_queue.LastModifiedTimestamp,
+            queue.maximum_message_size = sqs_queue.MaximumMessageSize,
+            queue.message_retention_period = sqs_queue.MessageRetentionPeriod,
+            queue.policy = sqs_queue.Policy, queue.arn = sqs_queue.QueueArn,
+            queue.receive_message_wait_time_seconds = sqs_queue.ReceiveMessageWaitTimeSeconds,
+            queue.redrive_policy_dead_letter_target_arn = sqs_queue.RedrivePolicy.deadLetterTargetArn,
+            queue.redrive_policy_max_receive_count = sqs_queue.RedrivePolicy.maxReceiveCount,
+            queue.visibility_timeout = sqs_queue.VisibilityTimeout,
+            queue.kms_master_key_id = sqs_queue.KmsMasterKeyId,
+            queue.kms_data_key_reuse_period_seconds = sqs_queue.KmsDataKeyReusePeriodSeconds,
+            queue.fifo_queue = sqs_queue.FifoQueue,
+            queue.content_based_deduplication = sqs_queue.ContentBasedDeduplication,
+            queue.deduplication_scope = sqs_queue.DeduplicationScope,
+            queue.fifo_throughput_limit = sqs_queue.FifoThroughputLimit,
+            queue.lastupdated = {aws_update_tag}
+        WITH queue
+        MATCH (owner:AWSAccount{id: {AWS_ACCOUNT_ID}})
+        MERGE (owner)-[r:RESOURCE]->(queue)
+        ON CREATE SET r.firstseen = timestamp()
+        SET r.lastupdated = {aws_update_tag}
+    """
+    dead_letter_queues: List[Dict] = []
+    queues: List[Dict] = []
+    for url, queue in data.items():
+        queue['url'] = url
+        queue['name'] = queue['QueueArn'].split(':')[-1]
+        queue['CreatedTimestamp'] = int(queue['CreatedTimestamp'])
+        queue['LastModifiedTimestamp'] = int(queue['LastModifiedTimestamp'])
+        dead_letter_arn = queue.get('RedrivePolicy', {}).get('deadLetterTargetArn')
+        if dead_letter_arn:
+            dead_letter_queues.append({
+                'arn': queue['QueueArn'],
+                'dead_letter_arn': dead_letter_arn,
+            })
+        queues.append(queue)
+
+    neo4j_session.run(
+        ingest_queues,
+        Queues=queues,
+        Region=region,
+        AWS_ACCOUNT_ID=current_aws_account_id,
+        aws_update_tag=aws_update_tag,
+    )
+
+    _attach_dead_letter_queues(neo4j_session, dead_letter_queues, aws_update_tag)
+
+
+@timeit
+def _attach_dead_letter_queues(neo4j_session: neo4j.Session, data: List[Dict[str, str]], aws_update_tag: int) -> None:
+    """
+    Attach deadletter queues to their queues.
+    """
+    attach_deadletter_to_queue = """
+    UNWIND {Relations} as relation
+        MATCH (queue:SQSQueue{id: relation.arn}), (deadletter:SQSQueue{id: relation.dead_letter_arn})
+        MERGE (queue)-[r:HAS_DEADLETTER_QUEUE]->(deadletter)
+        ON CREATE SET r.firstseen = timestamp()
+        SET r.lastupdated = {aws_update_tag}
+    """
+    neo4j_session.run(
+        attach_deadletter_to_queue,
+        Relations=data,
+        aws_update_tag=aws_update_tag,
+    )
```

Comment:
@achantavy can you please take a quick look at this code section to see if everything is okay?

Reply:
On a quick pass it looks good to me!